src/main/scala/is/hail/expr/ir/functions/UtilFunctions.scala

```diff
@@ -28,6 +28,11 @@ object UtilFunctions extends RegistryFunctions {
       ArrayMap(a, imul, ApplyBinaryPrimOp(Multiply(), Ref(imul, c.typ), c))
     }
 
+    registerIR("/", TArray(tv("T", t => t.isInstanceOf[TFloat32] || t.isInstanceOf[TFloat64])), tv("T")){ (a, c) =>
```

Comment:
I'm confused why you're restricting to TFLoat32 and TFloat64 versus `tnum`, since looking at the BinaryOp call we seem to support all four types.

Reply:
integers are promoted to floating point before doing division